### PR TITLE
Amend SAS token response

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[SAS token generated should include ? prefix](https://tools.hmcts.net/jira/browse/BPS-251)

### Change description ###

- Prepend with `?`
- as discussed, renaming some files to represent what data exactly is

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by rkondratowicz** [comment]
Are you sure this is actually a part of the token? or just what happens to be used to build the url (query params) in the shell script used for manual tests?
This endpoint was used before by our partner and it worked for them.
Will this change not break it?

**Comment by doncem** [comment]
No it will not break. Since palo alto or whatever it's name is the requirement of `?` is mandatory otherwise everything fails. Library used in the service generates same structure token as the one in Azure portal just without preceding `?`. Azure accepts both versions. palo - with `?`. The situation has been explained throughout number of meetings and this change is less than trivial to address it

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/SasQueryParamsGenerator.java [hunk 1]
```diff
@@ -24,30 +24,31 @@
...
 
         log.info("SAS Token request received for service {} ", serviceName);
 
         try {
-            return cloudBlobClient
+            String token = cloudBlobClient
                 .getContainerReference(serviceName)
                 .generateSharedAccessSignature(createSharedAccessPolicy(serviceName), null);
 
+            return token.charAt(0) == '?' ? token : "?" + token;
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/SasQueryParamsGenerator.java]: Are we covering test cases for both the scenarios ?

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/SasQueryParamsGenerator.java]: all tests written around sas token generator/controller are not using any mocks and token just comes back without `?`. Don't see the benefit of adding tests introducing mocks just to cover the branch despite the fact I'm in favour of having everything covered

**Comment by doncem** [comment]
Partners came back with explanation they are appending `?` before sticking in sas query params retrieved from our endpoint. The ticket becomes obsolete and therefore PR too. QA should be aware of this as well so similar _false negatives_ can be avoided 👍

**Pull request closed without merging**